app/src/main/java/awais/instagrabber/utils/KeywordsFilterUtils.java

```diff
@@ -0,0 +1,50 @@
+package awais.instagrabber.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import awais.instagrabber.repositories.responses.Caption;
+import awais.instagrabber.repositories.responses.Media;
+
+public final class KeywordsFilterUtils {
+
+    private final ArrayList<String> keywords;
+
+    public KeywordsFilterUtils(final ArrayList<String> keywords){
+        this.keywords = keywords;
+    }
+
+    public boolean filter(final String caption){
+        if(caption == null) return false;
+        if(keywords.isEmpty()) return false;
+        final String temp = caption.toLowerCase(Locale.getDefault());
+        for(final String s:keywords){
+            if(temp.contains(s)) return true;
+        }
+        return false;
+    }
+
+    public boolean filter(final Media media){
+        if(media == null) return false;
+        final Caption c = media.getCaption();
+        if(c == null) return false;
+        if(keywords.isEmpty()) return false;
+        final String temp = c.getText().toLowerCase(LocaleUtils.getCurrentLocale());
```

Comment:
Captions can be in any language though?

```suggestion
        final String temp = c.getText().toLowerCase();
```